net-p2p/hefur/hefur-0.6.recipe

```diff
@@ -0,0 +1,105 @@
+SUMMARY="A standalone and lightweight BitTorrent tracker"
+DESCRIPTION="Hefur is a standalone C++ BitTorrent tracker.
+
+Features:
+ - support HTTP, HTTPS and UPD announce and scrape
+ - support both IPv4 and IPv6
+ - no database required (everything in memory)
+ - should be efficient
+ - perfect for small setup
+ - info_hash white list based on .torrent in a directory tree
+ - control interface
+ - control program: hefurctl"
+HOMEPAGE="https://github.com/abique/hefur"
+COPYRIGHT="2012 Alexandre Bique"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/abique/hefur/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="ce693eb5f0410e47cd1f65906b77c9b0179eb807943119b900a8fa8f630d0d7b"
+SOURCE_FILENAME="hefur-$portVersion.tar.gz"
+PATCHES="hefur-$portVersion.patchset" # Merged upstream
+srcGitRev2="22137aa7f90b0f90bc48172d964deb82d6e4f2f9"
+SOURCE_URI_2="https://github.com/abique/mimosa/archive/$srcGitRev2.tar.gz"
+CHECKSUM_SHA256_2="abe7992fd45d9a458ebc8786de750ebbe6b961076065f4b6f6f7a870d524fc29"
+SOURCE_FILENAME_2="mimosa-$srcGitRev2.tar.gz"
+SOURCE_DIR_2="mimosa-$srcGitRev2"
+PATCHES_2="mimosa-$srcGitRev2.patchset" # Merged upstream
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ -z "${ARCHITECTURES##*!$targetArchitecture*}" ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	hefur$secondaryArchSuffix = $portVersion
+	cmd:hefurctl$commandSuffix = $portVersion
+	cmd:hefurd$commandSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix >= r1~alpha4_pm_hrev51768-1
```

Comment:
Please drop the version requirement in the REQUIRES section.
Haikuporter will use whatever version is used to build the package instead.

Reply:
`memmem(3)` is required, which had been added to `libgnu` in `hrev51768`. Since libgnu is dynamically linked, we need the newer version resides in `hrev51768` for the port to work.

Reply:
Sorry I must request once again. "Haikuporter will use whatever version is used to build the package instead."

Reply:
@alaviss, I think the reason why @korli is asking for the removal of the reference to a specific hrev is that having such a reference causes other problems.
(Anyone failing to build hefur will just need to update Haiku.)

Reply:
I think what @korli means is `>= r1~alpha4_pm_hrev51768-1` is added by HaikuPorter at packaging anyway (this happens to libs, so I believe the same holds for everything else).

Also, I've already enforced it in `BUILD_REQUIRES`, so user wouldn't be able to build it unless they have a new enough version.

Reply:
Thanks!